adafruit_circuitplayground/express.py

```diff
@@ -662,6 +662,52 @@ def stop_tone(self):
             self._sample = None
         self._speaker_enable.value = False
 
+    def play_melody(self, frequencies, durations, speed=10):
+        """ Play a melody using notes and beats, rests are 0 frequency.
+
+        :param int frequency: The frequency of the tone in Hz
+        :param float duration: The duration of the tone in seconds
+        :param float speed: Optional parameter, fine tune duration
+
+        .. image :: ../docs/_static/speaker.jpg
+          :alt: Onboard speaker
+
+        .. code-block:: python
+
+            from adafruit_circuitplayground.express import cpx
+
+            notes = [698, 880, 988, 698, 880, 988, 698, 880, 988,
+            1319, 1175, 988, 1047, 988, 784, 659, 587, 659, 784, 659,
+            698, 880, 988, 698, 880, 988, 698, 880, 988, 1319,
+            1175, 988, 1047, 1319, 988, 784, 988, 784, 587, 659,
+            587, 659, 698, 784, 880, 988, 1047, 988, 659, 698,
+            784, 880, 988, 1047, 1175, 1319, 1397, 1568,
+            587, 659, 698, 784, 880, 988, 1047, 988, 659,
+            784, 698, 880, 784, 988, 880, 1047, 988, 1175, 1047,
+            1319, 1175, 1397, 1319, 1319, 1397, 1175, 1319,
+            0, 0, 0]
+
+            beats= [2, 2, 4, 2, 2, 4, 2, 2, 2, 2, 4, 2, 2, 2, 2, 8, 2, 2, 2,
+            10, 2, 2, 4, 2, 2, 4, 2, 2, 2, 2, 4, 2, 2, 2, 2, 8, 2, 2, 2, 10,
+            2, 2, 4, 2, 2, 4, 2, 2, 8, 2, 2, 4, 2, 2, 4, 2, 2, 8,
+            2, 2, 4, 2, 2, 4, 2, 2, 8,
+            2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 2, 6,
+            4, 2, 2]
+
+            while True:
+                cpx.play_melody(notes, beats)
+        """
+        # Play a melody.
+        for idx, frequency in enumerate(frequencies):
+            if frequency is 0:
+                self._sample.stop()
+                time.sleep(durations[idx]/speed)
+            else:
+                self.start_tone(frequency)
+                time.sleep(durations[idx]/speed)
+                self._sample.stop()
```

Comment:
While `is 0` works, I would think equality comparison is better than membership comparison here. Also, if `self._sample.stop()` is always called no matter the `frequency`, you could shorten this.
```python
for idx, frequency in enumerate(frequencies):
    if frequency != 0:
        self.start_tone(frequency)
    time.sleep(durations[idx]/speed)
    self._sample.stop()
self._sample.stop()

Reply:
yes it is needed, if you dont call self._sample.stop() before self.start_tone() the new tone will not play

so you would like something like this instead?

```
for idx, frequency in enumerate(frequencies):
    if frequency != 0:
        self._sample.stop()
        self.start_tone(frequency)
    self._sample.stop()
    time.sleep(durations[idx]/speed)
    self._sample.stop()
self.stop_tone()
```

Reply:
yes this works very well thanks.